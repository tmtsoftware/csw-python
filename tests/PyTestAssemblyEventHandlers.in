{'_type': 'SystemEvent', 'eventId': 'test', 'source': 'CSW.TestPublisher', 'eventName': 'testEvent1', 'eventTime': {'seconds': 0, 'nanos': 0}, 'paramSet': [{'DoubleKey': {'keyName': 'assemblyEventValue', 'values': [42.0], 'units': 'NoUnits'}}]}
{'_type': 'SystemEvent', 'eventId': 'test', 'source': 'CSW.TestPublisher', 'eventName': 'testEvent2', 'eventTime': {'seconds': 0, 'nanos': 0}, 'paramSet': [{'IntKey': {'keyName': 'IntValue', 'values': [42], 'units': 'arcsec'}}, {'IntArrayKey': {'keyName': 'IntArrayValue', 'values': [[1, 2, 3, 4], [5, 6, 7, 8]], 'units': 'NoUnits'}}, {'FloatArrayKey': {'keyName': 'FloatArrayValue', 'values': [[1.2, 2.3, 3.4], [5.6, 7.8, 9.1]], 'units': 'marcsec'}}, {'IntMatrixKey': {'keyName': 'IntMatrixValue', 'values': [[[1, 2, 3, 4], [5, 6, 7, 8]], [[-1, -2, -3, -4], [-5, -6, -7, -8]]], 'units': 'meter'}}]}
{'_type': 'SystemEvent', 'eventId': 'test', 'source': 'CSW.TestPublisher', 'eventName': 'testEvent3', 'eventTime': {'seconds': 0, 'nanos': 0}, 'paramSet': [{'CoordKey': {'keyName': 'CoordParam', 'values': [{'_type': 'EqCoord', 'tag': 'BASE', 'ra': 659912249999, 'dec': -109892300000, 'frame': 'FK5', 'catalogName': 'none', 'pm': {'pmx': 0.5, 'pmy': 2.33}}, {'_type': 'SolarSystemCoord', 'tag': 'BASE', 'body': 'Venus'}, {'_type': 'MinorPlanetCoord', 'tag': 'GUIDER1', 'epoch': 2000, 'inclination': 324000000000, 'longAscendingNode': 7200000000, 'argOfPerihelion': 360000000000, 'meanDistance': 1.4, 'eccentricity': 0.234, 'meanAnomaly': 792000000000}, {'_type': 'CometCoord', 'tag': 'BASE', 'epochOfPerihelion': 2000.0, 'inclination': 324000000000, 'longAscendingNode': 7200000000, 'argOfPerihelion': 360000000000, 'perihelionDistance': 1.4, 'eccentricity': 0.234}, {'_type': 'AltAzCoord', 'tag': '301 deg', 'alt': 153000000000, 'az': 0}], 'units': 'NoUnits'}}, {'ByteKey': {'keyName': 'byteValue', 'values': b'\xde\xad\xbe\xef', 'units': 'NoUnits'}}, {'IntKey': {'keyName': 'IntValue', 'values': [42], 'units': 'arcsec'}}, {'FloatKey': {'keyName': 'floatValue', 'values': [42.1], 'units': 'arcsec'}}, {'LongKey': {'keyName': 'longValue', 'values': [42], 'units': 'arcsec'}}, {'ShortKey': {'keyName': 'shortValue', 'values': [42], 'units': 'arcsec'}}, {'BooleanKey': {'keyName': 'booleanValue', 'values': [True, False], 'units': 'arcsec'}}, {'ByteArrayKey': {'keyName': 'ByteArrayValue', 'values': [b'\xde\xad\xbe\xef', b'\x01\x02\x03\x04'], 'units': 'NoUnits'}}, {'IntArrayKey': {'keyName': 'IntArrayValue', 'values': [[1, 2, 3, 4], [5, 6, 7, 8]], 'units': 'NoUnits'}}, {'FloatArrayKey': {'keyName': 'FloatArrayValue', 'values': [[1.2, 2.3, 3.4], [5.6, 7.8, 9.1]], 'units': 'arcsec'}}, {'DoubleArrayKey': {'keyName': 'DoubleArrayValue', 'values': [[1.2, 2.3, 3.4], [5.6, 7.8, 9.1]], 'units': 'arcsec'}}, {'IntMatrixKey': {'keyName': 'IntMatrixValue', 'values': [[[1, 2, 3, 4], [5, 6, 7, 8]], [[-1, -2, -3, -4], [-5, -6, -7, -8]]], 'units': 'meter'}}, {'StructKey': {'keyName': 'MyStruct', 'values': [{'paramSet': [{'CoordKey': {'keyName': 'CoordParam', 'values': [{'_type': 'EqCoord', 'tag': 'BASE', 'ra': 659912249999, 'dec': -109892300000, 'frame': 'FK5', 'catalogName': 'none', 'pm': {'pmx': 0.5, 'pmy': 2.33}}, {'_type': 'SolarSystemCoord', 'tag': 'BASE', 'body': 'Venus'}, {'_type': 'MinorPlanetCoord', 'tag': 'GUIDER1', 'epoch': 2000, 'inclination': 324000000000, 'longAscendingNode': 7200000000, 'argOfPerihelion': 360000000000, 'meanDistance': 1.4, 'eccentricity': 0.234, 'meanAnomaly': 792000000000}, {'_type': 'CometCoord', 'tag': 'BASE', 'epochOfPerihelion': 2000.0, 'inclination': 324000000000, 'longAscendingNode': 7200000000, 'argOfPerihelion': 360000000000, 'perihelionDistance': 1.4, 'eccentricity': 0.234}, {'_type': 'AltAzCoord', 'tag': '301 deg', 'alt': 153000000000, 'az': 0}], 'units': 'NoUnits'}}, {'IntKey': {'keyName': 'IntValue', 'values': [42], 'units': 'arcsec'}}, {'FloatKey': {'keyName': 'floatValue', 'values': [42.1], 'units': 'arcsec'}}, {'LongKey': {'keyName': 'longValue', 'values': [42], 'units': 'arcsec'}}, {'ShortKey': {'keyName': 'shortValue', 'values': [42], 'units': 'arcsec'}}, {'BooleanKey': {'keyName': 'booleanValue', 'values': [True, False], 'units': 'arcsec'}}, {'IntArrayKey': {'keyName': 'IntArrayValue', 'values': [[1, 2, 3, 4], [5, 6, 7, 8]], 'units': 'NoUnits'}}, {'FloatArrayKey': {'keyName': 'FloatArrayValue', 'values': [[1.2, 2.3, 3.4], [5.6, 7.8, 9.1]], 'units': 'arcsec'}}, {'DoubleArrayKey': {'keyName': 'DoubleArrayValue', 'values': [[1.2, 2.3, 3.4], [5.6, 7.8, 9.1]], 'units': 'arcsec'}}, {'IntMatrixKey': {'keyName': 'IntMatrixValue', 'values': [[[1, 2, 3, 4], [5, 6, 7, 8]], [[-1, -2, -3, -4], [-5, -6, -7, -8]]], 'units': 'meter'}}]}], 'units': 'NoUnits'}}]}
