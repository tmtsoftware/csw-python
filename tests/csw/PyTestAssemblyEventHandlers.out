SystemEvent(testEvent1, assemblyEventValue((42.0)none))
SystemEvent(testEvent2, FloatArrayValue(((1.2,2.3,3.4),(5.6,7.8,9.1))mas), IntArrayValue(((1,2,3,4),(5,6,7,8))none), IntMatrixValue((((1,2,3,4),(5,6,7,8)),((-1,-2,-3,-4),(-5,-6,-7,-8)))m), IntValue((42)arcsec))
SystemEvent(testEvent3, ByteArrayValue(((-34,-83,-66,-17),(1,2,3,4))none), CoordParam((EqCoord(12:13:14.150  -30:31:32.300 FK5 none BASE 0.5/2.33),SolarSystemCoord(BASE,Venus),MinorPlanetCoord(GUIDER1,2000.0,Angle(90.0 degree),Angle(2.0 degree),Angle(100.0 degree),1.4,0.234,Angle(220.0 degree)),CometCoord(BASE,2000.0,Angle(90.0 degree),Angle(2.0 degree),Angle(100.0 degree),1.4,0.234),AltAzCoord(301 deg 42.5  0.0))none), DoubleArrayValue(((1.2,2.3,3.4),(5.6,7.8,9.1))arcsec), FloatArrayValue(((1.2,2.3,3.4),(5.6,7.8,9.1))arcsec), IntArrayValue(((1,2,3,4),(5,6,7,8))none), IntMatrixValue((((1,2,3,4),(5,6,7,8)),((-1,-2,-3,-4),(-5,-6,-7,-8)))m), IntValue((42)arcsec), booleanValue((true,false)none), byteValue((-34,-83,-66,-17)none), floatValue((42.1)arcsec), longValue((42)arcsec), shortValue((42)arcsec))
SystemEvent(testEvent4, assemblyEventValue((UTCTime(2021-09-20T20:43:35.419053077Z))none), assemblyEventValue2((TAITime(2021-09-20T18:44:12.419084072Z))none))
